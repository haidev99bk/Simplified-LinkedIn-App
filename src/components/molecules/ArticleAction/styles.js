import {StyleSheet} from 'react-native';
import fontWeight from '../../../styles/fontWeight';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-between',
    alignItems: 'center',
    height: 24,
    flex: 1,
  },
  iconContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 20,
    height: 20,
  },
  label: {
    ...fontWeight.semiBold,
  },
});

export default styles;
