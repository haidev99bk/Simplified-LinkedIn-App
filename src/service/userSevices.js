const UserAndFollowers = require('./mockData/users.json');

export function getUserInfo() {
  return UserAndFollowers.users[0];
}
