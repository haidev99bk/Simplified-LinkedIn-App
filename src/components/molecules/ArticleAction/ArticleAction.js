import {
  View,
  Text,
  TouchableWithoutFeedback,
  TouchableOpacity,
} from 'react-native';
import React from 'react';
import styles from './styles';
import Icon from '../../atoms/Icon';

const ArticleAction = props => {
  const {iconFamily, iconName, iconColor, label, onPress} = props;
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.container}>
        <View style={styles.iconContainer}>
          <Icon
            family={iconFamily}
            name={iconName}
            size={16}
            color={iconColor}
          />
        </View>
        <Text style={styles.label}>{label}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default React.memo(ArticleAction);
