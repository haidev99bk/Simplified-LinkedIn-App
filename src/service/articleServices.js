import {useId} from 'react';
import {getUserInfo} from './userSevices';
import AsyncStorage from '@react-native-async-storage/async-storage';

const Articles = require('./mockData/uArticles.json');

export function getArticles(userId) {
  const user = getUserInfo(userId);
  return Articles.articles.filter(a => {
    return user.articlesId.includes(a.articleId);
  });
}
