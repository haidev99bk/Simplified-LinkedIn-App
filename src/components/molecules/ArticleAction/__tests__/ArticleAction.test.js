import 'react-native';
jest.mock('react-native-device-info', () => {
  return {
    hasNotch: () => true,
  };
});
import React from 'react';
import renderer from 'react-test-renderer';
import {getArticles} from '../../../../service/articleServices';
import ArticleAction from '../ArticleAction';
import color from '../../../../styles/colors';
console.log('getArticles ==> ', getArticles(1), typeof getArticles(1));

test('FollowerArticles rendered correctly', () => {
  const snap = renderer
    .create(
      <ArticleAction
        iconFamily="AntDesign"
        iconName="like1"
        iconColor={color.mainBlue}
        label="Like"
        onPress={() => {}}
      />,
    )
    .toJSON();

  expect(snap).toMatchSnapshot();
});
