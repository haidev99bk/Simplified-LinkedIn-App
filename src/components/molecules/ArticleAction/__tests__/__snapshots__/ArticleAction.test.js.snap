// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowerArticles rendered correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "height": 24,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#1566c2",
              "fontSize": 16,
            },
            undefined,
            {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î™Œ
      </Text>
    </View>
    <Text
      style={
        {
          "fontWeight": "600",
        }
      }
    >
      Like
    </Text>
  </View>
</View>
`;
